src/account/AccountPickScreen.js

```diff
@@ -11,6 +12,10 @@ import AccountList from './AccountList';
 import { navigateToRealmScreen, accountSwitch, removeAccount } from '../actions';
 
 type Props = $ReadOnly<{|
+  // Since we've put this screen in a stack-nav route config, it gets the
+  // `navigation` prop for free.
```

Comment:
IIUC the precise logic is: because the *only* way we invoke this component is as a screen in such a route config, it will always get that prop.

Or I guess, equally correct and broader: because that's the only *non-type-checked* way we invoke it.

Typically I wouldn't expect us to use one of these components in any other way. But as a way of helping confirm that, I did this quick search: `rg -w '[A-Z]\w+Screen' src/` to find all references to things that fit the naming pattern common to almost all these components. Other than their own definitions, including react-redux `connect` calls, the only ones I found referred to anywhere other than `AppNavigator` were:
* `CompatibilityScreen`
* `ChooseRecipientsScreen`

And indeed those don't appear in this commit, so that checks out.

I repeated the search on the components that appear in this commit and don't fit that naming pattern: `rg -we SmartUrlInput -e MainScreenWithTabs -e MessageReactionList src/`

That actually found one reference that doesn't line up! Did you mean to include `SmartUrlInput`?

Reply:
Ah—SmartUrlInput should probably go in its own commit. It's a reusable component that expects...*a* `navigation` prop, which it doesn't get for free (from being plugged into a navigator's route config) but it shouldn't care whether that `navigation` prop comes from a stack navigator, drawer navigator, etc. I landed on `NavigationScreenProp<NavigationState>` without much consideration from https://github.com/react-navigation/react-navigation/issues/3643#issuecomment-370044086, and it could be wrong.